### PR TITLE
Translate body part

### PR DESCRIPTION
**Pull request by BenceVirtonomy**

### CODE REVIEW

File: cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp [hunk 1]
```diff
@@ -198,16 +198,16 @@ TEST(StructuralSimulation, TranslateSolidBodyTuple)
...
 	input.translation_solid_body_tuple_ = { TranslateSolidBodyTuple(0, end_time * 0.32, end_time, translation_vector) };
 
 	//=================================================================================================//
 	TestStructuralSimulation sim (input);
 	sim.TestRunSimulation(end_time);
 	//=================================================================================================//
 
-	StdLargeVec<Vecd>& pos_0 = sim.Get_translation_solid_body_()[0]->GetParticlePos0();
-	StdLargeVec<Vecd>& pos_n = sim.Get_translation_solid_body_()[0]->GetParticlePosN();
+StdLargeVec<Vecd>& pos_0 = sim.Get_solid_body_list_()[0].get()->getElasticSolidParticles()->pos_0_;
```
Comment (chuvirtonomy) [inline cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp:209]: what is the first get?

File: cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp [hunk 2]
```diff
@@ -216,6 +216,66 @@ TEST(StructuralSimulation, TranslateSolidBodyTuple)
...
+	};
+	Vecd translation_vector = Vec3d(0.0, 0.0, 0.02);
+	input.translation_solid_body_part_tuple_ = { TranslateSolidBodyPartTuple(0, end_time * 0.124, end_time, translation_vector, bbox) };
+
+	//=================================================================================================//
+	TestStructuralSimulation sim (input);
+	sim.TestRunSimulation(end_time);
+	//=================================================================================================//
+
+	StdLargeVec<Vecd>& pos_0 = sim.Get_solid_body_list_()[0].get()->getElasticSolidParticles()->pos_0_;
```
Comment (chuvirtonomy) [inline cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp:261]: get()?

File: SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp [hunk 3]
```diff
@@ -355,7 +356,7 @@ void StructuralSimulation::initializeAccelerationForBodyPartInBoundingBox()
 	{
 		SolidBody* solid_body = solid_body_list_[get<0>(acceleration_bounding_box_tuple_[i])]->getImportedModel();
 		acceleration_bounding_box_.emplace_back(make_shared<solid_dynamics::AccelerationForBodyPartInBoundingBox>
-			(solid_body, &get<1>(acceleration_bounding_box_tuple_[i]), get<2>(acceleration_bounding_box_tuple_[i])));
+			(solid_body, get<1>(acceleration_bounding_box_tuple_[i]), get<2>(acceleration_bounding_box_tuple_[i])));
```
Comment (chuvirtonomy) [inline SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp:359]: what is get<1> or get<2>

**Review by chuvirtonomy — APPROVED** [review]

File: SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp (hunk 3 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp:359]: to access the elements of the tuple

Comment (chuvirtonomy) [reply SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp:359]: ah ok

File: cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp (hunk 1 above)
Comment (BenceVirtonomy) [reply cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp:209]: to get the pointer to access the member
I can't access it directly

File: cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp (hunk 2 above)
Comment (BenceVirtonomy) [reply cases_high_level_simulation/test_3d_unit_position_based_bc/position_based_boundary_conditions.cpp:261]: ok

File: SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.cpp [hunk 4]
```diff
@@ -364,12 +364,32 @@ namespace SPH
 			}
 		}
 		//=================================================================================================//
+		bool checkIfPointInBoundingBox(Vecd point, BoundingBox& bbox)
```
Comment (Xiangyu-Hu) [inline SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.cpp:367]: This is not member function. Probably put behind the definition of bounding box?

Comment (Xiangyu-Hu) [reply SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.cpp:367]: Three functions each for 1, 2 and 3 d case?

File: SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.h [hunk 5]
```diff
@@ -241,7 +241,13 @@ namespace SPH
 			virtual void Update(size_t index_i, Real dt = 0.0) override;
 		};
 
-				/**
+		/**
```
Comment (Xiangyu-Hu) [inline SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.h:244]: Here seems not the place to declare this function.

File: SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.cpp (hunk 4 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.cpp:367]: okey, I will move it

File: SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.h (hunk 5 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.h:244]: okey

**Review by Xiangyu-Hu — APPROVED** [review]

**Pull request merged**